### PR TITLE
Port in hmm_posterior_sample function

### LINKED ISSUE #3: Port in dynamax hmm posterior inference
Use [`dynamax.hidden_markov_model.inference.hmm_posterior_sample`](https://github.com/probml/dynamax/blob/3aec95cfed3c2aeb84ff33d77639d23f91782f2d/dynamax/hidden_markov_model/inference.py#L487) in place of [`jax_moseq.utils.distributions.sample_hmm_stateseq`](https://github.com/dattalab/jax-moseq/blob/09542dfba208abf373dbb42682fabd0abf705de1/jax_moseq/utils/distributions.py#L46) (called by [`jax_moseq.models.arhmm.gibbs.resample_discrete_stateseqs`](https://github.com/dattalab/jax-moseq/blob/09542dfba208abf373dbb42682fabd0abf705de1/jax_moseq/models/arhmm/gibbs.py#L51))

## Approach
- Masking can be handled by pre-applying the mask directly to the log-likelihoods. For example, suppose we have an emission that we would like to ignore, i.e. `mask_t=0`. Then, in this proposal, `ll_t <- (log_likelihoods * mask)[t] = 0`. So, (see comments showing equivalence),
```
  def _forward_message(carry, args):
      ll_t, mask_t = args
      in_potential, logtot = carry
      cmax = ll_t.max()                                     # equals 0
      alphan_t = in_potential * jnp.exp(ll_t - cmax)        # jnp.exp(0 - 0) = 1 -> alphan_t = in_potential
      norm = alphan_t.sum() + 1e-16                         # norm = 1 since alphan_t = in_potential is already normalized
      alphan_t = alphan_t / norm
      logprob = jnp.log(norm) + cmax                        # jnp.log(1) = 0 -> logprob = 0 so doesn't contribute to logtot
      in_potential = alphan_t.dot(pi)*mask_t + in_potential*(1-mask_t)
                                                            # in_potential_next = alphan_t @ pi = in_potential @ pi
      return (in_potential, logtot + logprob*mask_t), alphan_t    
```

- Evaluate that the two implementations produce the same states in expectation
  - Generate a random sequence of HMM emissions and random binary mask, length $T$, $K$ states
  - For each implementation
    - Draw N samples
    - Compute empirical expected state for each time step, shape $T \times K$ 
  - The two empirical state distributions should be close

### PR DESCRIPTION
**Pull request by ezhang94**
Closes #3

### CODE REVIEW

File: jax_moseq/models/arhmm/gibbs.py [hunk 1]
```diff
@@ -47,11 +47,17 @@ def resample_discrete_stateseqs(seed, x, mask, Ab, Q, pi, **kwargs):
         Discrete state sequences.
     """
     nlags = get_nlags(Ab)
+
+    # TODO Why not use jax.vmap here? Can specify out_axes=-1 to remove
```
Comment (calebweinreb) [inline jax_moseq/models/arhmm/gibbs.py:51]: `jax.lax.map` is used to reduce peak VRAM usage, i.e. to prevent OOM errors for larger datasets.

**Review by calebweinreb — APPROVED** [review]

File: jax_moseq/models/arhmm/gibbs.py (hunk 1 above)
Comment (ezhang94) [reply jax_moseq/models/arhmm/gibbs.py:51]: Great to know!
